File: Step 3: Solve Problems on Arrays/Step 3.1: Easy/findMaxConsecutiveOnes.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int findMaxConsecutiveOnes(const vector<int>& nums) {
        int cnt = 0, ans = 0;
        for (size_t i = 0; i < nums.size(); i++) {
            cnt = (nums[i] == 1) ? cnt + 1 : 0;
            ans = max(ans, cnt);
        }
        return ans;
    }
};

int main() {
    vector<int> nums = {1, 1, 0, 1, 1, 1};
    Solution obj;
    int ans = obj.findMaxConsecutiveOnes(nums);
    cout << "The maximum  consecutive 1's are " << ans;
    return 0;
}

File: Step 3: Solve Problems on Arrays/Step 3.1: Easy/findMaxConsecutiveOnes.py
from typing import List


class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        ans = cnt = 0
        for i in range(0, len(nums)):
            cnt = cnt + 1 if nums[i] == 1 else 0
            ans = max(ans, cnt)
        return ans


obj = Solution()
print(obj.findMaxConsecutiveOnes(nums=[1, 1, 0, 1, 1, 1]))
print(obj.findMaxConsecutiveOnes(nums=[1, 0, 1, 1, 0, 1]))
